ethereum/spec/src/main/java/tech/pegasys/teku/spec/genesis/GenesisGenerator.java

```diff
@@ -102,11 +102,11 @@ public void updateCandidateState(
   }
 
   private void processActivation(final Deposit deposit) {
-    final Integer index = keyCache.get(deposit.getData().getPubkey());
-    if (index == null) {
-      // Could be null if the deposit was invalid
+    if (!keyCache.containsKey(deposit.getData().getPubkey())) {
+      // Could be absent if the deposit was invalid
       return;
     }
+    final int index = keyCache.get(deposit.getData().getPubkey());
```

Comment:
Given we can wind up running this in a tight loop it may be worth avoiding the additional map lookup by doing something like:

```suggestion
    final int index = keyCache.getOrDefault(deposit.getData().getPubkey(), -1);
    if (index == -1) {
      // Could be absent if the deposit was invalid
      return;
    }
```

We can't have negative deposit indices so -1 works fine as a sentinel value. Not a big performance difference but an easy win for no extra code complexity.

Reply:
Going to try committing this and merging - I have a terrible track record with breaking spotless when writing code in GitHub comments though so don't panic if you get failure notifications, I'll sort it... :)